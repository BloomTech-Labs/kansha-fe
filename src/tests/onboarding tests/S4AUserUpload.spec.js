import React from "react";
import { render,fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import S4UserUpload from '../../components/Onboarding/S4AUserUpload';
import {BrowserRouter} from 'react-router-dom'

let S4Component = <BrowserRouter><S4UserUpload /></BrowserRouter>;
test("S4UserUpload renders without crashing", () => {
  render(S4Component);
  
});

test('it displays add employees question', () => {
    const { getByText, getAllByText } = render(S4Component);
    getByText(/Would you like to add employees now?/i);
    getByText(/yes/i);
    getAllByText(/no/i);

})
test("the next button is clickable", () => {
    const button = jest.fn();
    const { getByText } = render(
        <BrowserRouter><S4UserUpload button={button}/></BrowserRouter>
    );
    const nextBtn = getByText(/next/i);
    fireEvent.click(nextBtn);
    expect(button).toHaveBeenCalled();
  });
  

//   const toggleLocked = jest.fn();
//   const { getByText } = render(
//     <Controls closed={false} toggleLocked={toggleLocked} />
//   );
//   const lockedBtn = getByText(/lock gate/i);
//   fireEvent.click(lockedBtn);
//   expect(toggleLocked).not.toHaveBeenCalled();
// });