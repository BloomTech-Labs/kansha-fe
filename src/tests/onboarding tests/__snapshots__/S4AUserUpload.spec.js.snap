// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`S4 matches snapshot 1`] = `
Array [
  <div
    className="admin-user-add-container"
  >
    <h2
      className="user-upload-title"
    >
      Add Employees
    </h2>
    <div
      className="admin-add-employee-form"
    >
      <div
        className="upload-now-question"
      >
        <h6>
          Would you like to add employees now?
        </h6>
        <div>
          <input
            id="yes"
            name="drone"
            onChange={[Function]}
            type="radio"
            value="true"
          />
          <label
            htmlFor="yes"
          >
            Yes
          </label>
        </div>
        <div>
          <input
            id="no"
            name="drone"
            onChange={[Function]}
            type="radio"
            value="false"
          />
          <label
            htmlFor="no"
          >
            No
          </label>
        </div>
      </div>
    </div>
  </div>,
  <button
    onClick={[Function]}
  >
    Next
  </button>,
  <div
    className="step-p-container"
  >
    <span
      className="previousarrow"
    >
      <i
        className="fas fa-arrow-left"
      />
      <div
        data-testid="button"
        onClick={[Function]}
      >
        <p>
           Previous step
        </p>
      </div>
    </span>
  </div>,
  <p>
    Continue later
  </p>,
]
`;
